sdk/compiler/damlc/tests/src/DA/Test/DamlcUpgrades.hs

```diff
@@ -298,17 +298,22 @@ tests damlc =
               "FailsWithSynonymReturnTypeChangeInSeparatePackage"
               (FailWithError "\ESC\\[0;91merror type checking template Main.T choice C:\n  The upgraded choice C cannot change its return type.")
               LF.versionDefault
-              SeparateDeps
+              (SeparateDeps False)
         , test
               "SucceedsWhenUpgradingADependency"
               Succeed
               LF.versionDefault
-              SeparateDeps
+              (SeparateDeps False)
         , test
               "FailsOnlyInModuleNotInReexports"
               (FailWithError "\ESC\\[0;91merror type checking data type Other.A:\n  The upgraded data type A has added new fields, but those fields are not Optional.")
               LF.versionDefault
               NoDependencies
+        , test
+              "FailsWhenDepsDowngradeVersions"
+              (FailWithError "\ESC\\[0;91merror type checking <none>:\n  Dependency upgrades-example-FailsWhenDepsDowngradeVersions-dep has version 0.0.1 on the upgrading package, which is older than version 0.0.2 on the upgraded package.\n  Dependency versions of upgrading packages must always be greater or equal to the dependency versions on upgraded packages.")
+              LF.versionDefault
+              (SeparateDeps True)
```

Comment:
We add a new test for the simplest case. SeparateDeps now takes a flag to tell it to flip the versions of dependencies. TODO: Maybe more thorough testing would be good?

Reply:
Ticket created: https://github.com/digital-asset/daml/issues/19283